server/src/main/kotlin/io/provenance/invoice/services/EventHandlerService.kt

```diff
@@ -90,21 +89,28 @@ class EventHandlerService(
             .use { signatureStream ->
                 val messageBytes = signatureStream.readAllBytes()
                 val targetInvoice = Asset.parseFrom(messageBytes).unpackInvoiceI()
-                try {
+                Either.catch {
```

Comment:
This, on the other hand, shows how you can have better flow-control with `Either`, which is sweet